### PR TITLE
Users can check out#31

### LINKED ISSUE #42: User Story 31, Registered users can check out
```
As a registered user
When I add items to my cart
And I visit my cart
I see a button or link indicating that I can check out
And I click the button or link to check out
An order is created in the system, which has a status of "pending"
I am taken to my orders page ("/profile")
I see a flash message telling me my order was created
I see my new order listed on my profile orders page
My cart is now empty
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #49

child of #79

depends on #47

### PR DESCRIPTION
**Pull request by timnallen**
As a registered user
When I add items to my cart
And I visit my cart
I see a button or link indicating that I can check out
And I click the button or link to check out
An order is created in the system, which has a status of "pending"
I am taken to my orders page ("/profile")
I see a flash message telling me my order was created
I see my new order listed on my profile orders page
My cart is now empty

Closes #42 

Guys, this one is kinda crazy... I think we should go over it together tomorrow before anyone merges it. Ultimately, in order to checkout, I needed to change how we were organizing the contents of cart. Instead of a hash with a string id => string quantity, I had to pass in a hash as the value with the quantity and unit price in it (otherwise, I couldn't create order_items, which require a unit_price value).

I'll explain more tomorrow. I fixed everything this broke in my machine, but there may be more in unmerged PRs. (It took a LONG time because apparently symbols get turned into strings sometimes...? So, I just made the keys strings in the first place to avoid that problem.

Edit: just talked to Peter about a possible easier work around that is limited to the orders#create action where we can find the item from the I'd and access the price right there

### CODE REVIEW

File: app/views/carts/show.html.erb [hunk 1]
```diff
@@ -49,9 +49,9 @@
 
 
     <div class="col-xs-2">
-      <button type="button" class="btn btn-success btn-block">
-        Checkout
-      </button>
+      <%= button_to "Checkout", profile_orders_path %>
+      <!-- <button type="button" class="btn btn-success btn-block">
+      </button> -->
           </div>
```
Comment (timnallen) [inline app/views/carts/show.html.erb]: No clue how to put these together, so I broke the nice looking green button...

File: app/views/orders/index.html.erb [hunk 2]
```diff
@@ -4,4 +4,5 @@
   <%= order.updated_at %>
   <%= order.quantity_of_items %>
   <%= order.grand_total %>
+  <%= order.status %>
 <% end %>
```
Comment (timnallen) [inline app/views/orders/index.html.erb:8]: Can't tell if the user story needs this, but I put it here to pass a test that we may need to change.

Comment (timnallen) [reply app/views/orders/index.html.erb:8]: Re:  status

**Review by teresa-m-knowles — APPROVED** [review]
We reviewed this together

**Review by csvlewis — APPROVED** [review]
We just went over this together, looks good!

**Review by allieIsOkay — APPROVED** [review]
Reviewed together. Approved

**Pull request merged**